aiohttp/http_parser.py

```diff
@@ -727,30 +727,36 @@ def flush(self) -> bytes:
             self.decompressor = BrotliDecoder()  # type: Any
         else:
             zlib_mode = (16 + zlib.MAX_WBITS
-                         if encoding == 'gzip' else -zlib.MAX_WBITS)
+                         if encoding == 'gzip' else zlib.MAX_WBITS)
             self.decompressor = zlib.decompressobj(wbits=zlib_mode)
 
     def set_exception(self, exc: BaseException) -> None:
         self.out.set_exception(exc)
 
     def feed_data(self, chunk: bytes, size: int) -> None:
+        if not size:
+            return
+
         self.size += size
+
+        # RFC1950
+        # bits 0..3 = CM = 0b1000 = 8 = "deflate"
+        # bits 4..7 = CINFO = 1..7 = windows size.
+        if not self._started_decoding and self.encoding == 'deflate' \
```

Comment:
@asvetlov should we decode non-RFC compliant data? I would remove such support. But unfortunately, this will break communication with deflate-enabled aiohttp server.

Reply:
Let's keep it for a while. I have a feeling that we need to support this mode for three years at least.

Reply:
`urllib3` used by `requests` has a similar workaround: https://github.com/urllib3/urllib3/blob/9971e27e83a891ba7b832fa9e5d2f04bbcb1e65f/src/urllib3/response.py#L55-L61